.github/workflows/install_hdf5.sh

```diff
@@ -0,0 +1,10 @@
+# Download and install HDF5 1.10.7 from source for building wheels
+curl https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-1.10/hdf5-1.10.7/src/hdf5-1.10.7.tar.gz --output hdf5-1.10.7.tar.gz --silent
+tar -xvf hdf5-1.10.7.tar.gz
```

Comment:
@james-s-willis one concern I have is about if things will work with mismatched versions, e.g. you're building the wheels against HDF5 1.10, but the library installed on a system might be a 1.8 series one, which should in theory be installed. Is there any way you could add a test to the CI which checks that the binary wheels will work against a 1.8 series HDF5?

Reply:
I think if you can run the tests twice, you might be able to install HDF5 via yum (provided it's a new enough 1.8 series), and reset LD_LIBRARY_PATH and run it.

Reply:
@jrs65, so the problem with using the HDF5 that comes with yum is that it is HDF5 1.8.5, where bitshuffle needs the 1.8.11 for dynamically loaded filters. So I can try building against 1.8.11 and see if that works.